packages/walletkit/src/new-web3-utils.ts

```diff
@@ -0,0 +1,144 @@
+import {
+  Callback,
+  ErrorCallback,
+  JSONRPCRequestPayload,
+  Subprovider,
+  Web3ProviderEngine,
+} from '@0x/subproviders'
+import debugFactory from 'debug'
+import Web3 from 'web3'
+import { JsonRPCResponse, Provider } from 'web3/providers'
+import { CeloProvider } from './transaction-utils'
+
+const debug = debugFactory('kit:web3:utils')
+
+export function generateAccountAddressFromPrivateKey(privateKey: string): string {
+  if (!privateKey.toLowerCase().startsWith('0x')) {
+    privateKey = '0x' + privateKey
+  }
+  return new Web3().eth.accounts.privateKeyToAccount(privateKey).address
+}
+
+// This web3 has signing ability.
+export function addLocalAccount(web3: Web3, privateKey: string): Web3 {
+  const celoProvider = new CeloProvider(privateKey)
+  const existingProvider = web3.currentProvider
+  let providerEngine: Web3ProviderEngine
+  if (existingProvider instanceof Web3ProviderEngine) {
+    debug('Existing provider is already a Web3ProviderEngine')
```

Comment:
here we use debug but in that other file we use Logger. we should probably stay consistent

Reply:
I backported this file from `contractkit`. Replacing `debug` with `Logger.debug`